Foreign/LibFFI/Base.hs

```diff
@@ -49,6 +49,23 @@ newStructCType cTypes = do
     init_ffi_type ffi_type elements
     return (ffi_type, free ffi_type >> free elements)
 
+sizeAndAlignmentOfCType :: Ptr CType -> IO (Int, Int)
```

Comment:
In addition to providing this function, I think it would be worth adding a similar function of type `RetType a -> IO (Int, Int)`, since that would be more convenient to use in combination with [`newStorableStructArgRet`](https://hackage.haskell.org/package/libffi-0.2/docs/src/Foreign.LibFFI.Base.html#newStorableStructArgRet).

Reply:
Makes sense, I'll add that too then

Reply:
Actually, looking at it, since `RetType` doesn't actually directly contain the `CType`, I don't see an elegant way of implementing what you're asking for. Either the definition of `RetType` would have to be changed, or a new function like `newStorableStructArgRet` could be added that also returns the size and alignment. But I just realized, that if you're using something like `newStorableStructArgRet` then you should already know the size and alignment, since that's [part of the `Storable` class](https://hackage.haskell.org/package/base-4.17.0.0/docs/Foreign-Storable.html#t:Storable). 

For some context, I implemented this feature because I needed it for my language runtime. Since the language is dynamic, FFI struct types are just represented byte arrays passed in. Because of the dynamic nature, the size and alignment is only know at the time of the call, so I can't use `Storable`, and hence why I needed this API.

Reply:
> But I just realized, that if you're using something like `newStorableStructArgRet` then you should already know the size and alignment, since that's [part of the `Storable` class](https://hackage.haskell.org/package/base-4.17.0.0/docs/Foreign-Storable.html#t:Storable).

A fair point—I had overlooked the fact that `Storable` gives you this already. In that case, this API is fine.